app/src/androidTestGecko/java/org/mozilla/focus/activity/ImageSelectTest.java

```diff
@@ -0,0 +1,238 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.activity;
+
+import android.content.Context;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.test.uiautomator.UiObject;
+import android.support.test.uiautomator.UiObjectNotFoundException;
+import android.support.test.uiautomator.UiSelector;
+import android.view.KeyEvent;
+
+import junit.framework.Assert;
+
+import org.junit.After;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import org.mozilla.focus.helpers.TestHelper;
+
+import static android.support.test.espresso.action.ViewActions.click;
+import static org.junit.Assert.assertTrue;
+import static org.mozilla.focus.helpers.TestHelper.waitingTime;
+import static org.mozilla.focus.fragment.FirstrunFragment.FIRSTRUN_PREF;
+
+@RunWith(AndroidJUnit4.class)
+public class ImageSelectTest {
+    private static final String TEST_PATH = "/";
+
+    private Context appContext;
+    private MockWebServer webServer;
+
+    @Rule
+    public ActivityTestRule<MainActivity> mActivityTestRule  = new ActivityTestRule<MainActivity>(MainActivity.class) {
+        @Override
+        protected void beforeActivityLaunched() {
+            super.beforeActivityLaunched();
+
+            appContext = InstrumentationRegistry.getInstrumentation()
+                    .getTargetContext()
+                    .getApplicationContext();
+
+            PreferenceManager.getDefaultSharedPreferences(appContext)
+                    .edit()
+                    .putBoolean(FIRSTRUN_PREF, true)
+                    .apply();
+
+            webServer = new MockWebServer();
+
+            try {
+                webServer.enqueue(new MockResponse()
+                        .setBody(TestHelper.readTestAsset("image_test.html"))
+                        .addHeader("Set-Cookie", "sphere=battery; Expires=Wed, 21 Oct 2035 07:28:00 GMT;"));
+                webServer.enqueue(new MockResponse()
+                        .setBody(TestHelper.readTestAsset("rabbit.jpg")));
```

Comment:
\[PMD\] INFO: The String literal "rabbit\.jpg" appears 4 times in this file; the first occurrence is on line 67